scripts/sysdig-deploy/update-sysdig-deploy.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+set -e
+
+node_analyzer_chart_path="node-analyzer"
+agent_chart_path="agent"
+kspm_collectore_chart_path="kspm-collector"
+rapid_response_chart_path="rapid-response"
+admission_controller_chart_path="admission-controller"
+sysdig_deploy_path="sysdig-deploy"
+
+minor=0
+major=0
+patch=0
+
+check_update_needed () {
+    
+    new_subchart_version=$(yq eval '.version' charts/$chart/Chart.yaml)
+    sysdig_subchart_version=$(yq eval '.dependencies[] | select(.name == "'$chart'") | .version ' charts/$sysdig_deploy_path/Chart.yaml | sed "s/~//")
+
+    echo "subchart version in" $chart"/Chart.yaml"
+    echo $new_subchart_version
+    echo $chart "subchart version in sysdig-deploy/Chart.yaml"
+    echo $sysdig_subchart_version
+
+
+    IFS="."
+    read -ra prev_ver <<< "$sysdig_subchart_version"
+    read -ra next_ver <<< "$new_subchart_version"
```

Comment:
I guess this works perfectly fine as is -- still, I'd try to avoid any subtle side effects by setting IFS locally:
```suggestion
    IFS="." read -ra prev_ver <<< "$sysdig_subchart_version"
    IFS="." read -ra next_ver <<< "$new_subchart_version"
```